airflow/utils/dag_processing.py

```diff
@@ -0,0 +1,657 @@
+                self._last_runtime[file_path] = (now -
+                                                 processor.start_time).total_seconds()
+                self._last_finish_time[file_path] = now
+                self._run_count[file_path] += 1
+            else:
+                running_processors[file_path] = processor
+        self._processors = running_processors
+
+        # Collect all the DAGs that were found in the processed files
+        simple_dags = []
+        for file_path, processor in finished_processors.items():
+            if processor.result is None:
+                self.logger.warn("Processor for {} exited with return code "
+                                 "{}. See {} for details."
+                                 .format(processor.file_path,
+                                         processor.exit_code,
+                                         processor.log_file))
+            else:
+                for simple_dag in processor.result:
+                    simple_dags.append(simple_dag)
+
+        # Generate more file paths to process if we processed all the files
+        # already.
+        if len(self._file_path_queue) == 0:
+            # If the file path is already being processed, or if a file was
+            # processed recently, wait until the next batch
+            file_paths_in_progress = self._processors.keys()
+            now = datetime.now()
+            file_paths_recently_processed = []
+            for file_path in self._file_paths:
+                last_finish_time = self.get_last_finish_time(file_path)
+                if (last_finish_time is not None and
+                    (now - last_finish_time).total_seconds() <
+                        self._process_file_interval):
+                    file_paths_recently_processed.append(file_path)
+
+            files_paths_at_run_limit = [file_path
+                                        for file_path, num_runs in self._run_count.items()
+                                        if num_runs == self._max_runs]
+
+            files_paths_to_queue = list(set(self._file_paths) -
+                                        set(file_paths_in_progress) -
+                                        set(file_paths_recently_processed) -
+                                        set(files_paths_at_run_limit))
+
+            for file_path, processor in self._processors.items():
+                self.logger.debug("File path {} is still being processed (started: {})"
+                                  .format(processor.file_path,
+                                          processor.start_time.isoformat()))
+
+            self.logger.debug("Queuing the following files for processing:\n\t{}"
+                              .format("\n\t".join(files_paths_to_queue)))
+
+            self._file_path_queue.extend(files_paths_to_queue)
+
+        # Start more processors if we have enough slots and files to process
+        while (self._parallelism - len(self._processors) > 0 and
```

Comment:
just want to flag that the scheduler will likely need to have a decently high parallelism (the `max_threads` setting in the scheduler config, defaulting to the cpu count) since we have several dags that take a lot of time to process.